### PR TITLE
Fix Broken Links

### LINKED ISSUE #2602: Please help us fix broken links!
free-programming-books-pt_PT.md

free-programming-books-de.md

free-programming-books-cs.md

free-programming-books-ua.md

free-programming-books-se.md

free-programming-books-hu.md

free-programming-books-pt_BR.md
+ [x] 500 Can't connect to metamorphoselinux.net:80 http://metamorphoselinux.net/Shell_Script_do_Zero.pdf

free-programming-interactive-tutorials-en.md

free-podcasts-screencasts-es.md

free-podcasts-screencasts-cs.md

free-podcasts-screencasts-en.md

free-programming-books-ru.md

free-programming-books-bg.md

free-programming-books-sk.md

free-programming-books-pl.md

free-podcasts-screencasts-ru.md

free-programming-playgrounds.md
+ [x] 500 Can't connect to phpfiddle.org:80 http://phpfiddle.org

free-programming-books-it.md

free-podcasts-screencasts-se.md

free-programming-books-fi.md

free-courses-ru.md

free-programming-books-ro.md

free-programming-books-tr.md

problem-sets-competitive-programming.md
+ [ ] 500 Can't connect to acm.hdu.edu.cn:80 http://acm.hdu.edu.cn
+ [ ] 404 Not Found http://problembook.net/content/en/index.html

free-programming-books-fa_IR.md

free-programming-books-gr.md

free-programming-books-fr.md

javascript-frameworks-resources-pt_BR.md

free-programming-books.md
+ [x] 500 Can't connect to jeffe.cs.illinois.edu:80 http://jeffe.cs.illinois.edu/teaching/algorithms/
+ [x] 500 Can't connect to mfleck.cs.illinois.edu:80 http://mfleck.cs.illinois.edu/building-blocks/index.html
+ [ ] 500 Can't connect to blog.hit.edu.cn:80 http://blog.hit.edu.cn/jsx/upload/AT%EF%BC%86TAssemblyLanguage.pdf
+ [x] 500 Can't connect to www.slideshare.net:443 http://www.slideshare.net/olvemaudal/deep-c
+ [x] 503 Service Unavailable https://web.archive.org/web/20161220072449/http://www.brpreiss.com/books/opus6/
+ [ ] 521 Origin Down http://rypress.com/tutorials/git
+ [ ] 404 Not Found http://docs.spring.io/spring/docs/current/spring-framework-reference/pdf/spring-framework-reference.pdf
+ [ ] 404 Not Found http://health.adelaide.edu.au/psychology/ccs/teaching/lsr

free-programming-books-es.md
+ [ ] 404 Not Found http://library.originalhacker.org/biblioteca/articulo/ver/123

free-programming-books-az.md
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/c-proqramlasdirma-dili
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/css
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/html
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/javascript
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/linux
+ [x] 502 Bad Gateway http://ilkaddimlar.com/ders/php

free-programming-books-ja.md
+ [ ] 404 Not Found https://developer.apple.com/jp/devcenter/ios/library/documentation/OpenGLES_ProgrammingGuide.pdf
+ [ ] 404 Not Found http://www.oki-osk.jp/esc/groovy/
+ [ ] 404 Not Found http://www.fos.kuis.kyoto-u.ac.jp/~t-sekiym/classes/isle4/mltext/ocaml.html
+ [ ] 404 Not Found http://download.bnn.co.jp/download/beyond_interaction/BNN_BeyondInteraction.pdf

free-programming-books-ko.md

free-podcasts-screencasts-pt_BR.md

javascript-frameworks-resources.md
+ [ ] 500 Can't connect to reactjs.net:80 http://reactjs.net/getting-started/aspnet5.html

free-programming-books-id.md
+ [ ] 403 Forbidden http://www.ilmuwebsite.com/ebook-php-free-download

free-courses-en.md

free-programming-books-ar.md

free-programming-books-mm.md

free-programming-books-zh.md
+ [ ] 302 Found http://neo4j.tw
+ [x] 502 Bad Gateway http://www.gruntjs.net
+ [ ] 500 Can't connect to deerchao.net:80 http://deerchao.net/tutorials/regex/regex.htm
+ [ ] 404 Not Found https://github.com/Flowerowl/Big-Data-Resources
+ [ ] 500 Can't connect to deerchao.net:80 http://deerchao.net/tutorials/regex/regex.htm
+ [ ] 500 Can't connect to local.joelonsoftware.com:80 http://local.joelonsoftware.com/wiki/Chinese_(Simplified)
+ [ ] 500 Can't connect to local.joelonsoftware.com:80 http://local.joelonsoftware.com/wiki/%E9%A6%96%E9%A0%81
+ [ ] 500 Status read failed: Connection reset by peer http://sewm.pku.edu.cn/src/paradise/reference/CMake%20Practice.pdf
+ [ ] 404 Not Found http://typeof.net/s/jsmech/
+ [ ] 404 Not Found http://learningcn.com/underscore/
+ [ ] 500 Status read failed: Connection reset by peer http://net.pku.edu.cn/~yhf/tutorial/perl/perl.html
+ [ ] 500 Can't connect to jwch.sdut.edu.cn:80 http://jwch.sdut.edu.cn/book/rst.html

### PR DESCRIPTION
**Pull request by JacksonIsaac**
Resolves #2602 

Remove broken links. Few links were working which were tagged as not reachable with 500 code. Replaced outdated links or websites that are down with the ones from web.archive.org.

### CODE REVIEW

File: free-programming-books-az.md [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 ### HTML
 
-* [HTML](http://ilkaddimlar.com/ders/html)
+* [HTML](https://web.archive.org/web/20170606111135/http://ilkaddimlar.com/ders/html)
```
[the remark this answers is not in the document]
Comment (JacksonIsaac) [reply free-programming-books-az.md]: On my network, it was unable to find the website. Probably due to regional restriction or something. Although it loads on my mobile network. Reverting back the change for this link.

File: free-programming-books-ja.md [hunk 2]
```diff
@@ -477,15 +476,10 @@
 
 ### OCaml
 
-* [Objective Caml 入門](http://www.fos.kuis.kyoto-u.ac.jp/~t-sekiym/classes/isle4/mltext/ocaml.html) - 五十嵐淳
+* [Objective Caml 入門](https://web.archive.org/web/20161128072705/http://www.fos.kuis.kyoto-u.ac.jp:80/~t-sekiym/classes/isle4/mltext/ocaml.html) - 五十嵐淳
 * [お気楽 OCaml プログラミング入門](http://www.geocities.jp/m_hiroi/func/ocaml.html) - 広井誠
 
 
-### openFrameworks
```
Comment (eshellman) [inline free-programming-books-ja.md]: is there a directory entry that needs removing?

File: free-programming-books-zh.md [hunk 3]
```diff
@@ -171,7 +170,7 @@
 
 * [Spark 编程指南简体中文版](https://aiyanbo.gitbooks.io/spark-programming-guide-zh-cn/content/)
 * [大型集群上的快速和通用数据处理架构](https://code.csdn.net/CODE_Translation/spark_matei_phd)
-* [大数据/数据挖掘/推荐系统/机器学习相关资源](https://github.com/Flowerowl/Big-Data-Resources)
+* [大数据/数据挖掘/推荐系统/机器学习相关资源](https://web.archive.org/web/20150326064941/https://github.com/Flowerowl/Big-Data-Resources)
```
Comment (eshellman) [inline free-programming-books-zh.md]: as this was not a resource but a list of resources, no longer being maintained, perhaps its better to just delete it.

File: free-programming-books-ja.md (hunk 2 above)
Comment (JacksonIsaac) [reply free-programming-books-ja.md]: There was only one link under openFrameworks which no longer works. Hence I removed it since it was empty.

**Pull request merged**